interp/notation_ops.ml

```diff
@@ -1104,8 +1137,9 @@ let match_termlist match_fun alp metas sigma rest x y iter termin revert =
     try
       let metas = add_ldots_var (add_meta_term y metas) in
       let (terms,_,_,_ as sigma) = match_fun metas sigma rest iter in
-      let rest = Id.List.assoc ldots_var terms in
-      let t = Id.List.assoc y terms in
+      let _,rest = Id.List.assoc ldots_var terms in
+      (* The vars are the same for all terms, we can drop them *)
```

Comment:
Just for clarity, what enforces this invariant?

Reply:
The vars are the same because the binders in each patterns (if there are some) have to match. However, thanks for the question: this is not a reason to drop the vars as we can have notations of the form:
```
Notation "[ x | y ; z ; .. ; t ]" := (pair .. (pair (forall x, y) .. (forall x, z) .. (forall x, t)).
```
or even the more complicated
```
Notation "[ x & p | y ; .. ; z ; t ]" := (forall x, p -> y -> .. (forall x, p -> z -> forall x, p -> t) ..).
```
I pushed a fix.